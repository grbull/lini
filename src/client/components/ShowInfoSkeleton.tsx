import cn from 'classnames';
import React, { ReactElement } from 'react';

import { InfoCard } from './InfoCard';
import { Skeleton } from './Skeleton';

export function ShowInfoSkeleton(): ReactElement {
  return (
    <>
      <InfoCard className={cn('my-0')}>
        <Skeleton
          className={cn('w-1/3', 'h-0', 'relative', 'float-left')}
          style={{ paddingTop: `${140 / 3}%` }}
        />
        <div className={cn('float-right', 'w-2/3', 'pl-2.5')}>
          <Skeleton className={cn('h-44')} />
        </div>
      </InfoCard>

      <InfoCard>
        <Skeleton className={cn('h-60')} />
      </InfoCard>

      <InfoCard>
        <Skeleton className={cn('h-32')} />
      </InfoCard>
    </>
  );
}
