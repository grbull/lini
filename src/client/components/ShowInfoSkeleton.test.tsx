/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import React from 'react';

import { ShowInfoSkeleton } from './ShowInfoSkeleton';

function setup(): RenderResult {
  const utils = render(<ShowInfoSkeleton />);

  return { ...utils };
}

describe('ShowInfoSkeleton Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
