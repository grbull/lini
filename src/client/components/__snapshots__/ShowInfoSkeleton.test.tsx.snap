// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowInfoSkeleton Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden my-0"
  >
    <div
      class="animate-pulse bg-gray-700 w-1/3 h-0 relative float-left"
      style="padding-top: 46.666666666666664%;"
    />
    <div
      class="float-right w-2/3 pl-2.5"
    >
      <div
        class="animate-pulse bg-gray-700 h-44"
      />
    </div>
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <div
      class="animate-pulse bg-gray-700 h-60"
    />
  </div>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    <div
      class="animate-pulse bg-gray-700 h-32"
    />
  </div>
</DocumentFragment>
`;
